main.py

```diff
@@ -283,16 +290,74 @@ def disable_switch(self, request: Request) -> JSONResponse:
         dpid = request.path_params["dpid"]
         try:
             switch = self.controller.switches[dpid]
+            link_ids = set()
+            for _, interface in switch.interfaces.copy().items():
+                if (interface.link and interface.link.is_enabled()):
+                    link_ids.add(interface.link.id)
+                    interface.link.disable()
+                    self.notify_link_enabled_state(interface.link, "disabled")
+            self.topo_controller.bulk_disable_links(link_ids)
```

Comment:
Notice that if an empty set is passed and `bulk_write(ops)` can raise `InvalidOperation`. 

Probably the easiest route here is to add an early return on `bulk_disable_links` for that case. Let's cover this input with an unit test too.

Reply:
Added statement to detect empty sets in `bulk_disable_links()`